### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by aliaksei-kosyrau-leverx**

### CODE REVIEW

**Comment by nalkhovikleverx** [comment]
The implementation should be moved from api to `internal/interfaces/inprocess` package. The api package should store just contracts, not an implementation.

File: internal/registration/api/inprocess/registration/registration.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	registrationHandler,
+	confirmationHandler application.CommandHandler,
+) *RegistrationModuleFacade {
+	return &RegistrationModuleFacade{
+		registrationHandler: registrationHandler,
+		confirmationHandler: confirmationHandler,
+	}
+}
+
+func CreateNewInProcessRegistrationModuleFacade() *RegistrationModuleFacade {
```
Comment (nalkhovikleverx) [inline internal/registration/api/inprocess/registration/registration.go:28]: You should create a `module.go` inside `GoEdu/internal/registration` package and build all the module dependencies in there for now. The example is in `GoEdu/internal/api/http` module.

**Review by nalkhovikleverx — CHANGES_REQUESTED** [review]

File: internal/registration/internal/infrastructure/inprocess/repository/memory/repository.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		i.store[snapShot.ID] = snapShot
+		return nil
+	}
+	return errors.New("this user doesn't exist")
+}
+
+func (i *InProcessRepository) GetAll(_ context.Context) []domain.UserRegistration {
+	temp := make([]domain.UserRegistration, 0, len(i.store))
+	for _, val := range i.store {
+		temp = append(temp, *domain.MustCreateUserRegistrationFromSnapshot(val))
```
Comment (nalkhovikleverx) [inline internal/registration/internal/infrastructure/inprocess/repository/memory/repository.go:50]: return Snapshot

**Review by nalkhovikleverx — CHANGES_REQUESTED** [review]

File: internal/api/http/controller/registration.go [hunk 3]
```diff
@@ -1,34 +1,67 @@
...
+) *registrationHTTPController {
+	return &registrationHTTPController{tracer: t, registrationAPI: rAPI}
 }
 
 func (c *registrationHTTPController) RegistrationNew(w http.ResponseWriter, r *http.Request) {
 	logger.FromContext(r.Context()).Info("registration new")
-	w.WriteHeader(http.StatusNotImplemented)
+	cmd := inprocess.RegisterNewUserCommand{}
+
+	err := json.NewDecoder(r.Body).Decode(&cmd)
```
Comment (forgoty) [inline internal/api/http/controller/registration.go:30]: in `server.gen.go` there is `NewRegistration` request body struct. Please decode the json body into this struct and then map the `RegisterNewUserCommand` manually.

File: internal/registration/api/inprocess/interface.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package inprocess
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+type RegisterNewUserCommand struct {
+	FirstName string `json:"firstName"`
```
Comment (forgoty) [inline internal/registration/api/inprocess/interface.go:10]: remove `json` struct tags from this struct. All consumers of this API for now should map it manualy.

**Review by forgoty — CHANGES_REQUESTED** [review]

**Review by nalkhovikleverx — CHANGES_REQUESTED** [review]

File: internal/api/http/controller/mapper.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"GoEdu/internal/api/http/server"
+	"GoEdu/internal/registration/api/inprocess"
+)
+
+func FromRequestToRegisterNewUserCommand(registration server.NewRegistration) inprocess.RegisterNewUserCommand {
+	return inprocess.RegisterNewUserCommand{
+		FirstName: registration.FirstName,
+		LastName:  registration.LastName,
+		Email:     string(registration.Email),
```
Comment (nalkhovikleverx) [inline internal/api/http/controller/mapper.go:12]: `Password` field not mapped.

**Review by nalkhovikleverx — CHANGES_REQUESTED** [review]

**Review by nalkhovikleverx — APPROVED** [review]

**Pull request merged**